kpatch-build/create-diff-object.c

```diff
@@ -922,6 +1020,9 @@ static void kpatch_replace_sections_syms(struct kpatch_elf *kelf)
 				continue;
 			}
 
+#ifdef __powerpc__
+			add_off = 0;
+#else
```

Comment:
If I remember correctly, there are a few cases on x86 where `-ffunction-sections` doesn't work and certain functions aren't bundled into their own sections.  For example, functions in `.text.unlikely`.

So you might want to at least add a sanity check and ERROR() if the relocation type is `R_PPC64_REL24` and the if (rela->sym->sec && rela->sym->sec->sym) is false, to catch such a case.